package.json

```diff
@@ -39,10 +39,11 @@
     "docker:tag": "docker tag saucelabs/speedo saucelabs/speedo:$npm_package_version",
     "eslint": "eslint ./src ./tests",
     "prepublishOnly": "npm prune && run-s build",
-    "release": "npm run release:patch",
-    "release:patch": "np patch",
-    "release:minor": "np minor",
-    "release:major": "np major",
+    "release": "release-it",
+    "release:ci": "npm run release -- --ci --npm.skipChecks --no-git.requireCleanWorkingDir",
```

Comment:
why do we skip NPM checks and don't require a clean working dir?

Reply:
When publishing package in CI (e.g. Github Actions) we need to skip NPM checks, because this command doesn't respect credentials provided in `.npmrc` file create by CI, see [this document](https://github.com/release-it/release-it/blob/2b02a634e9292b57cc97a837b5a8e1fadeb4fa22/docs/ci.md#npm) for more details.

As for "don't require a clean working dir" rule, we need to skip it, because in the job we build new version of the package. I assumed that we don't want to commit it separately from version bump commit. Otherwise, we need to add a separate step to commit bundled version.

Reply:
from what I understand is we should make the token available in `.npmrc` via:

```sh
$ echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc
```

I don't see that it is fine just to have an `NODE_AUTH_TOKEN` environment variable set.

> because in the job we build new version of the package. I assumed that we don't want to commit it separately from version bump commit. Otherwise, we need to add a separate step to commit bundled version.

I expect that we would see a version bump commit after the pipeline as well as a release with a changelog.

Reply:
So step `actions/setup-node@v1` creates a `.npmrc` file, where `NODE_AUTH_TOKEN` is used as `NPM_TOKEN` in your example (basically [this line](https://github.com/actions/setup-node/blob/main/src/authutil.ts#L48)).

> I expect that we would see a version bump commit after the pipeline as well as a release with a changelog.

Yes, that's how it's going to work. Additionally, workflow will put a new bundled version (changed files in `./build` directory) in a commit with new version.

Reply:
> So step `actions/setup-node@v1` creates a `.npmrc`

Ah cool, but then we don't need to skip the check given that we have the token and the `.npmrc` file available, no?

Reply:
Unfortunately, `release-it` runs `npm whoami`, which requires to be logged in (via `npm login`) and ignores token. That's why we need to skip npm checks to not have authorization issues 😞

Reply:
Ah ok, got it 👍